test/functional/api/version_spec.lua

```diff
@@ -93,6 +93,7 @@ describe("api metadata", function()
   local api, compat, stable, api_level
   local old_api = {}
   setup(function()
+    clear() -- this should called to create a session before trying to request api_info
```

Comment:
why? there is a clear on line 50

Reply:
because `setup()` gets called first

> You can also use the functions `before_each` and `after_each` to define functions that should be run before any nested tests, and `setup` and `teardown` for functions that should be run before and after the `describe` block.

try this

```diff
diff --git a/test/functional/api/version_spec.lua b/test/functional/api/version_spec.lua
index 390c9a20f..645afc036 100644
--- a/test/functional/api/version_spec.lua
+++ b/test/functional/api/version_spec.lua
@@ -47,7 +47,10 @@ end)
 
 
 describe("api metadata", function()
-  before_each(clear)
+  before_each(function()
+    clear()
+    eq(nil, _G.api_methods)
+  end)
 
   local function name_table(entries)
     local by_name = {}
@@ -93,6 +96,7 @@ describe("api metadata", function()
   local api, compat, stable, api_level
   local old_api = {}
   setup(function()
+    _G.api_methods = 2
     clear() -- this should called to create a session before trying to request api_info
     api = meths.get_api_info()[2]
     compat  = api.version.api_compatible
```

here are the results
```console
$ TEST_FILE=test/functional/api/version_spec.lua cmake --build build --target functionaltest
[----------] Global test environment setup.
[----------] Running tests from test/functional/api/version_spec.lua
[ RUN      ] api_info()['version'] returns API level: 6.71 ms OK
[ RUN      ] api_info()['version'] returns Nvim version: 6.58 ms OK
nan ms test/helpers.lua:73: Expected objects to be the same.
Passed in:
(number) 2
Expected:
(nil)
```